qualtran/bloqs/arithmetic/addition.py

```diff
@@ -151,9 +171,12 @@ def _right_building_block(self, inp, out, anc, depth):
             return
         else:
             yield CNOT().on(anc[depth - 1], anc[depth])
-            yield And().adjoint().on(inp[depth], out[depth], anc[depth])
-            yield CNOT().on(anc[depth - 1], inp[depth])
-            yield CNOT().on(inp[depth], out[depth])
+            if depth < len(inp):
+                yield And().adjoint().on(inp[depth], out[depth], anc[depth])
+                yield CNOT().on(anc[depth - 1], inp[depth])
```

Comment:
what happens if `depth = 0`?

Reply:
depth starts off = 1

Reply:
please add a gaurd on the `depth - 1` statements to make the code safer, e.g.

```py3
if depth:
   yield CNOT().on(anc[depth - 1], inp[depth])
```

Reply:
What about adding an assert statement at the beginning of the _left_building_block? Because adding an if statement every time I have anc[depth - 1] will make the code kinda messy in my opinion.

Reply:
okay but make it more informative

```py3
if depth < 1:
   raise ValueError(f"{depth=} is not a positive integer")
```

Reply:
Ok I added a more informative message along with the assert